siege_utilities/geo/spatial_data.py

```
@@ -1010,8 +1040,15 @@ def validate_download_url(self, url: str) -> bool:
         except BoundaryUrlValidationError:
             raise
         except requests.exceptions.SSLError as ssl_err:
-            log.debug(f"SSL verification failed for {url}, trying without verification...")
             ctx["ssl_error"] = str(ssl_err)
+            if not _ALLOW_INSECURE_SSL:
+                log.error(
+                    "SSL verification failed for %s; refusing the "
+                    "verify=False fallback (set SIEGE_INSECURE_SSL=1 "
+                    "to opt in).", url,
+                )
```

Ruff (0.15.12)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)